enterprise/utils.py

```diff
@@ -2306,3 +2306,27 @@ def get_default_invite_key_expiration_date():
     The default expiration date is 365 days from the current date.
     """
     return localized_utcnow() + datetime.timedelta(days=365)
+
+
+def hide_price_when_zero(enterprise_customer, course_modes):
+    """
+    Adds a "hide_price" flag to the course modes if price is zero and "Hide course price when zero" flag is set.
+
+    Arguments:
+        enterprise_customer: The EnterpriseCustomer that the enrollemnt is being done.
+        course_modes: iterable with dictionaries containing a required 'final_price' key
+    """
+    if not enterprise_customer.hide_course_price_when_zero:
+        return course_modes
+
+    for mode in course_modes:
+        mode['hide_price'] = False
+        try:
+            numbers = re.findall(r'\d+', mode['final_price'])
+            mode['hide_price'] = int(''.join(numbers)) == 0
+        except ValueError:
```

Comment:
This can happen only when `mode['original_price']` does not contain digits (`mode['final_price']` is generated in `get_course_final_price`).
Technically, it could also happen when the `mode['min_price']` does not contain digits, but this should raise an exception in the `format_price` function.

Is it even possible? If yes, what does it mean? Is the course free? We can keep this exception handling; I'm just curious.

Reply:
@Agrendalath To be fair, I didn't quite find an edge case while testing for this. The final `final_price` is a formatted string from [`format_price`](https://github.com/open-craft/edx-enterprise/blob/62189afec3e8f9a211747d423b49da8f8a3d6190/enterprise/utils.py#L1019) which is bound to either provide a properly formatted string, or throw an exception. 

But for some reason, I wasn't comfortable with parsing string with regext with no safeguards, so added the try block.